bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -769,32 +771,35 @@ std::pair<PrePrepareMsg *, bool> ReplicaImp::finishAddingRequestsToPrePrepareMsg
           numOfTransientPreprepareMsgs_++;
         } else {
           LOG_ERROR(GL, "The number of transient preprepare messages are very large so they will have to be retried.");
-          delete prePrepareMsg;
           return std::make_pair(nullptr, false);
         }
+        // prePrepareMsg is about to be moved. save here values for future use
+        auto seqNum = prePrepareMsg->seqNumber();
+        auto cid = prePrepareMsg->getCid();
+        auto requestsSize = prePrepareMsg->requestsSize();
+        auto numberOfRequests = prePrepareMsg->numberOfRequests();
         threadPool.async(
-            [](auto *ppm, auto *iq, auto *hist) {
+            [](auto ppm, auto *iq, auto *hist) {
               {
                 TimeRecorder scoped_timer(*(hist->finishAddingRequestsToPrePrepareMsg));
                 ppm->finishAddingRequests();
               }
-              iq->pushInternalMsg(std::move(ppm));
+              iq->pushInternalMsg(ppm.release());
```

Comment:
I am not sure what the original writer meant to do here? ppm is a pointer. Why move a pointer? looks redundent.
do we have a task to convert pushInternalMsg and its internal mechanisms into smart pters?
I think no.

Reply:
Yes, I created one: [BC-26653](https://jira.eng.vmware.com/browse/BC-26653)